HermesProxy/World/Server/PacketHandlers/ItemHandler.cs

```diff
@@ -186,5 +186,13 @@ void HandleSetAmmo(SetAmmo ammo)
             packet.WriteUInt32(ammo.ItemId);
             SendPacketToServer(packet);
         }
+
+        [PacketHandler(Opcode.CMSG_CANCEL_TEMP_ENCHANTMENT)]
+        void HandleCancelTempEnchantment(CancelTempEnchantment cancel)
+        {
+            WorldPacket packet = new WorldPacket(Opcode.CMSG_CANCEL_TEMP_ENCHANTMENT);
```

Comment:
This opcode doesn't exist in vanilla. The assert in packet class will trigger and Hermes will crash when you attempt to send it. You should put a check for the LegacyVersion above, and return if its below 2.0.1.

Reply:
The "AddedInVersion" / "RemovedInVersion" syntax is exceptionally confusing to me. Would this here work too:

```cs
        void HandleCancelTempEnchantment(CancelTempEnchantment cancel)
        {
            if (GetSession().Build < (uint)ClientVersionBuild.V2_0_1_6180)
                return;
            WorldPacket packet = new WorldPacket(Opcode.CMSG_CANCEL_TEMP_ENCHANTMENT);
            packet.WriteUInt32(cancel.EnchantmentSlot);
            SendPacketToServer(packet);
        }
```

Reply:
AddedInVersion means >=
RemovedInVersion means <

Reply:
But what was removed?
Isn't it rather that it just wasn't added yet?
"IncludedUntil" / "IncludedSince" make more sense to me personally. :(

I hope the newest commits fixes the remaining issues! :)

Reply:
We are using the build checks from WPP. They are named like that there. This makes it easy to copy packet handlers from WPP to Hermes.

Reply:
fair enough